localstack/services/awslambda/lambda_api.py

```diff
@@ -1489,9 +1489,7 @@ def get_function(function):
     region = LambdaRegion.get()
     funcs = do_list_functions()
     for func in funcs:
-        if function == func["FunctionName"]:
-            return lookup_function(func, region, request.url)
-        elif function in func["FunctionArn"]:
```

Comment:
Thanks for this PR @jarmokoivuniemi . 👍  

The reason we have this check is for cases where `function` is a function ARN that contains a version specifier, or a partial ARN - see details in the original issue here: https://github.com/localstack/localstack/issues/3954

Could we try changing the logic to something like this:
```
   arn_regex = r".*%s($|:.+)" % function
   is_arn = ":" in function
   for func in funcs:
        if function == func["FunctionName"] or (is_arn and re.match(arn_regex, func["FunctionArn"])):
            return lookup_function(func, region, request.url)
```

Reply:
Regex is not really my thing but I copy-pasted your code and added a test. Seems to work 😁